增加了指定文件/目录的编码/解码功能

execute233 opened the pull request:
使用了多一行的代码，实现了更多的功能。

execute233 committed:
实现了更多的功能

execute233 committed:
增加跨平台兼容性，格式化代码.

deepsource-autofix[bot] committed:
style: format code with dotnet-format

Format code with dotnet-format

This commit fixes the style issues introduced in 6bfbf23 according to the output
from dotnet-format.

Details: None

execute233 committed:
Merge pull request #1 from execute233/deepsource-transform-a6685378

format code with dotnet-format

execute233 commented:
？？？

lc6464 commented:
我 LC6464.Base16384 都还没发正式版，你这个急什么？

execute233 commented:
6

lc6464 commented:
你先帮我把 main 的剩下的 public 方法测了吧😑

execute233 commented:
你不是已经preview了吗？赶紧升级为release吧。

execute233 commented:
除了有关Stream的方法都测过了。因为有关Stream的方法不知道是我的问题还是你的问题。

lc6464 commented:
文件开头有个 0xFE 0xFF，不参与解码的，你是不是没有去掉啊？

execute233 commented:
有没有种可能，你的DecodeFromFileToNewFile已经有了？

execute233 commented:
咋还不合并呢？

lc6464 commented:
类库 v2.0.0-RC.1 已发布。
记得测试，测试完成即可转正式版。
转完正式版即可审核此 PR.

lc6464 commented:
你也可以考虑在 Shared 最后加上 CopyToHGlobal 或者 CopyToNewHGlobal 之类的，不过我觉得没必要。

lc6464 reviewed (changes requested):
其实我的建议是，
``` pwsh
Base16384.Net.exe <"e" | "d"> <source | "-"> <out | "-">
```
"-" 代表 Standard IO
source 若为文件，则直接处理，输出文件名为 out。
source 若为文件夹，则按我一开始的方案遍历并输出到文件夹 out。

lc6464 commented:
> 其实我的建议是，
> ``` pwsh
> Base16384.Net.exe <"e" | "d"> <source | "-"> <out | "-">
> ```
> "-" 代表 Standard IO
> source 若为文件，则直接处理，输出文件名为 out。
> source 若为文件夹，则按我一开始的方案遍历并输出到文件夹 out。

另外，如果 out 为 stdout，则不做任何 Console.WriteLine 自然语言提醒，出现什么问题直接 return (not 0)。

execute233 commented:
-_-

lc6464 commented:
你先看看能不能帮我把那个修了吧🤔

execute233 commented:
6

execute233 commented:
Standard IO？stdout？有关这些的不会写。-_-

lc6464 commented:
你先把那个 UTF8 的问题修了吧。

execute233 commented:
有没有种可能，我已经发PR了？

execute233 commented:
> 其实我的建议是，
> 
> ```powershell
> Base16384.Net.exe <"e" | "d"> <source | "-"> <out | "-">
> ```
> 
> "-" 代表 Standard IO source 若为文件，则直接处理，输出文件名为 out。 source 若为文件夹，则按我一开始的方案遍历并输出到文件夹 out。

输出文件打算都改名为 源文件+.En/Decode

lc6464 commented:
你都写了这么多了再加一个处理很难吗？😑

execute233 commented:
> 其实我的建议是，
> 
> ```powershell
> Base16384.Net.exe <"e" | "d"> <source | "-"> <out | "-">
> ```
> 
> "-" 代表 Standard IO source 若为文件，则直接处理，输出文件名为 out。 source 若为文件夹，则按我一开始的方案遍历并输出到文件夹 out。

输出文件名应该是原文件名+".out"吧？

execute233 commented:
我觉得应该往Console输出提示文件到哪里了,也可以提示是否成功。

lc6464 commented:
> 输出文件名应该是原文件名+".out"吧？

你是真没理解还是假没理解啊？
输出文件名由用户决定。

lc6464 commented:
> 我觉得应该往Console输出提示文件到哪里了,也可以提示是否成功。

如果不是 Std out 那就输出，如果是，那就不输出。

execute233 commented:
万一用户懒得写输出文件名呢？

lc6464 commented:
> 万一用户懒得写输出文件名呢？

在输入文件名的基础上添加 .encoded/.decoded

execute233 commented:
standard IO真不知道咋写 --_--

execute233 commented:
standard IO真不知道咋写 --_--

lc6464 commented:
OpenStandardIn

execute233 commented:
standard IO真不知道咋写 --_--

execute233 committed:
写了点代码

deepsource-autofix[bot] committed:
style: format code with dotnet-format

This commit fixes the style issues introduced in 4ff3c14 according to the output
from dotnet-format.

Details: None

execute233 committed:
Merge pull request #2 from execute233/deepsource-transform-59dd27be

style: format code with dotnet-format

lc6464 commented:
Your ReadMe needs to be perfected.
And, some parts of code, such as file opening, may can be refined.

lc6464 commented:
I will fully review them when I come back to my dormitory.

lc6464 reviewed (changes requested):
Because of the code of poor quality and the readme, it can not be merged.
But thank you for your PR.
If you still want to contribute this project, please optimize your code, thank you.

execute233 committed:
-
